File: 品味Java的21種設計模式/dp_cpp/mediator/example1/ConcreteMediator.cpp
#include "ConcreteMediator.h"

namespace cn {
namespace javass {
namespace dp {
namespace mediator {
namespace example1 {

void ConcreteMediator::setConcreteColleagueA(ConcreteColleagueA *colleague) {
    colleagueA = colleague;
}

void ConcreteMediator::setConcreteColleagueB(ConcreteColleagueB *colleague) {
    colleagueB = colleague;
}

void ConcreteMediator::changed(Colleague *colleague) {
    //某个同事类发生了变化，通常需要与其他同事交户
    //具体协调相应的同事对象来实现协作行为
}
}
}
}
}
}

File: 品味Java的21種設計模式/dp_java/mediator/example1/ConcreteMediator.java
package cn.javass.dp.mediator.example1;
/**
 * 具体的中介者实现
 */
public class ConcreteMediator implements Mediator {

    /**
     * 持有并维护同事A
     */
    private ConcreteColleagueA colleagueA;
    /**
     * 持有并维护同事B
     */
    private ConcreteColleagueB colleagueB;

    /**
     * 设置中介者需要了解并维护的同事A对象
     * @param colleague 同事A对象
     */
    public void setConcreteColleagueA(ConcreteColleagueA colleague) {
        colleagueA = colleague;
    }
    /**
     * 设置中介者需要了解并维护的同事B对象
     * @param colleague 同事B对象
     */
    public void setConcreteColleagueB(ConcreteColleagueB colleague) {
        colleagueB = colleague;
    }

    public void changed(Colleague colleague) {
        //某个同事类发生了变化，通常需要与其他同事交户
        //具体协调相应的同事对象来实现协作行为
    }

}
